File: sylva-app/src/main/java/io/github/ryosuke37/sylva/controller/dto/PostDto.java
package io.github.ryosuke37.sylva.controller.dto;

import lombok.Value;
import lombok.With;

import java.time.LocalDateTime;

@Value
@With
public class PostDto {
    String id;
    String content;
    UserDto user;
    PostDto rootPost;
    PostDto parentPost;
    PostDto quotedPost;
    LocalDateTime createdDate;
    LocalDateTime updatedDate;
}

File: frontend/src/types/PostDto.ts
import { type UserDto } from "./UserDto";

export interface PostDto {
  id: string;
  content: string;
  user: UserDto;
  rootPost?: PostDto | null;
  parentPost?: PostDto | null;
  quotedPost?: PostDto | null;
  createdDate: string;
  updatedDate: string;
}
